### PR TITLE
python3Packages.pytimeparse2: init at 1.7.1 python3Packages.mysql-connector: 8.0.29 -> 8.0.33 sqlite3-to-mysql: 1.4.19 -> 2.0.3 

### PR DESCRIPTION
**Pull request by gador**
## Description of changes

For the update of `sqlite3-to-mysql` I had to update `mysql-connector` and init `pytimeparse2`:

- Changelog for `mysql-connector`: https://github.com/mysql/mysql-connector-python/blob/8.0.33/CHANGES.txt
  - For the "C" extension to work I had to add the corresponding libraries to `setup.py`. 
  - I removed "Allow for clang to be used to build native extensions" patch, since upstream incoprorated the change [here](https://github.com/mysql/mysql-connector-python/commit/6eb19b9efdca13df9b67417578b8390a2786928a)
- `pytimeparse2` : https://github.com/onegreyonewhite/pytimeparse2
- `sqlite3-to-mysql` Changelog: https://github.com/techouse/sqlite3-to-mysql/blob/master/CHANGELOG.md

Info at maintainers: @neosimsim @turion 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
Result of `nixpkgs-review pr 252135` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>python310Packages.mysql-connector</li>
    <li>python310Packages.mysql-connector.dist</li>
    <li>python310Packages.pytimeparse2</li>
    <li>python310Packages.pytimeparse2.dist</li>
    <li>python311Packages.mysql-connector</li>
    <li>python311Packages.mysql-connector.dist</li>
    <li>python311Packages.pytimeparse2</li>
    <li>python311Packages.pytimeparse2.dist</li>
    <li>sqlite3-to-mysql</li>
    <li>sqlite3-to-mysql.dist</li>
  </ul>
</details>

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 252135` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
10 packages built:
</summary>
<ul>
<li>python310Packages.mysql-connector</li>
<li>python310Packages.mysql-connector.dist</li>
<li>python310Packages.pytimeparse2</li>
<li>python310Packages.pytimeparse2.dist</li>
<li>python311Packages.mysql-connector</li>
<li>python311Packages.mysql-connector.dist</li>
<li>python311Packages.pytimeparse2</li>
<li>python311Packages.pytimeparse2.dist</li>
<li>sqlite3-to-mysql</li>
<li>sqlite3-to-mysql.dist</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
python310Packages.pytimeparse2.dist:
<details>

A typo in pythonImportsCheck got detected.

</details>
python310Packages.pytimeparse2:
<details>

A typo in pythonImportsCheck got detected.

</details>
python311Packages.pytimeparse2.dist:
<details>

A typo in pythonImportsCheck got detected.

</details>
python311Packages.pytimeparse2:
<details>

A typo in pythonImportsCheck got detected.

</details>
</details>

File: pkgs/development/python-modules/pytimeparse2/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  # custom checks, see
+  # https://github.com/onegreyonewhite/pytimeparse2/blob/e00df7506b6925f2c6a5783e89e9f239d128271a/tox.ini#L36C20-L36C78
+  checkPhase = ''
+    runHook preCheck
+    python tests.py -vv --failfast
+    runHook postCheck
+  '';
+
+  pythonImportChecks = [ "pytimepase" ];
```
Comment (turion) [inline pkgs/development/python-modules/pytimeparse2/default.nix:24]: Certainly a typo?

File: pkgs/development/python-modules/mysql-connector/default.nix [hunk 2]
```diff
@@ -30,17 +39,19 @@ buildPythonPackage rec {
     # 10.12. The patch reverts
     # https://github.com/mysql/mysql-connector-python/commit/d1e89fd3d7391084cdf35b0806cb5d2a4b413654
```
[the remark this answers is not in the document]
Comment (turion) [reply pkgs/development/python-modules/mysql-connector/default.nix:40]: Ah... apologies, the comment is probably still right.

**Review by turion — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/pytimeparse2/default.nix (hunk 1 above)
Comment (gador) [reply pkgs/development/python-modules/pytimeparse2/default.nix:24]: oh boy, actually two typos. Its `pythonImportsCheck` and not `pythonImportChecks`. This is why it didn't catch the error in the first place.

File: pkgs/development/python-modules/mysql-connector/0001-Revert-Fix-MacOS-wheels-platform-tag.patch [hunk 3]
```diff
@@ -1,36 +1,22 @@
...
 -    get_config_vars()["MACOSX_DEPLOYMENT_TARGET"] = "11.0"
--
- COMMON_USER_OPTIONS = [
-     ("byte-code-only", None,
-      "remove Python .py files; leave byte code .pyc only"),
--- 
-2.31.1
 
+ COMMON_USER_OPTIONS = [
+     (
```
[the remark this answers is not in the document]
Comment (gador) [reply pkgs/development/python-modules/mysql-connector/0001-Revert-Fix-MacOS-wheels-platform-tag.patch:22]: No, this is how upstream has its file formatted. See https://github.com/mysql/mysql-connector-python/blob/3fe87a163de1c1dfcf1de60bbabecc30394c3ef6/cpydist/__init__.py#L71-L75

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 252135` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
10 packages built:
</summary>
<ul>
<li>python310Packages.mysql-connector</li>
<li>python310Packages.mysql-connector.dist</li>
<li>python310Packages.pytimeparse2</li>
<li>python310Packages.pytimeparse2.dist</li>
<li>python311Packages.mysql-connector</li>
<li>python311Packages.mysql-connector.dist</li>
<li>python311Packages.pytimeparse2</li>
<li>python311Packages.pytimeparse2.dist</li>
<li>sqlite3-to-mysql</li>
<li>sqlite3-to-mysql.dist</li>
</ul>
</details>

**Review by turion — APPROVED** [review]

**Pull request merged**